### PR TITLE
Spring Solution

### PR DESCRIPTION
**Pull request by serg-ksv**

### CODE REVIEW

File: src/main/java/core/ksv/controller/CarController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @GetMapping
+    public List<CarResponseDto> getAll() {
+        return carService.getAll().stream()
+                .map(carMapper::getDtoFromCar)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/car")
```
Comment (Sofasmile) [inline src/main/java/core/ksv/controller/CarController.java]:
```suggestion
    @GetMapping("/{id}")
```

File: src/main/java/core/ksv/controller/CarController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    public CarResponseDto getById(@RequestParam Long id) {
+        var car = carService.getById(id);
+        return carMapper.getDtoFromCar(car);
+    }
+
+    @PostMapping("/update-car")
```
Comment (Sofasmile) [inline src/main/java/core/ksv/controller/CarController.java]:
```suggestion
    @PutMapping("/{id}")
```

File: src/main/java/core/ksv/controller/CarController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public CarResponseDto updateCar(@RequestParam Long id,
+                                    @RequestBody CarRequestDto requestDto) {
+        var car = carMapper.getCarFromDto(requestDto);
+        car.setId(id);
+        return carMapper.getDtoFromCar(carService.update(car));
+    }
+
+    @GetMapping("/delete-car")
```
Comment (Sofasmile) [inline src/main/java/core/ksv/controller/CarController.java]:
```suggestion
    @DeleteMapping("/{id}")
```

File: src/main/java/core/ksv/controller/CarDoorController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @GetMapping
+    public List<CarDoorResponseDto> getAll() {
+        return carDoorService.getAll().stream()
+                .map(carDoorMapper::getDtoFromCarDoor)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/car-door")
```
Comment (Sofasmile) [inline src/main/java/core/ksv/controller/CarDoorController.java]: the same

File: src/main/java/core/ksv/controller/CarWheelController.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @GetMapping
+    public List<CarWheelResponseDto> getAll() {
+        return carWheelService.getAll().stream()
+                .map(carWheelMapper::getDtoFromCarWheel)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/car-wheel")
```
Comment (Sofasmile) [inline src/main/java/core/ksv/controller/CarWheelController.java]: :)

Comment (serg-ksv) [reply src/main/java/core/ksv/controller/CarWheelController.java]: These suggestions make sense :)
Thank you

File: src/main/java/core/ksv/controller/CarDoorController.java (hunk 4 above)
Comment (serg-ksv) [reply src/main/java/core/ksv/controller/CarDoorController.java]: done

File: src/main/java/core/ksv/controller/CarController.java (hunk 3 above)
Comment (serg-ksv) [reply src/main/java/core/ksv/controller/CarController.java]: done

File: src/main/java/core/ksv/controller/CarController.java (hunk 2 above)
Comment (serg-ksv) [reply src/main/java/core/ksv/controller/CarController.java]: done

File: src/main/java/core/ksv/controller/CarController.java (hunk 1 above)
Comment (serg-ksv) [reply src/main/java/core/ksv/controller/CarController.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**